base/binaryplatforms.jl

```diff
@@ -873,7 +873,7 @@ function detect_libstdcxx_version(max_minor_version::Int=30)
     end
 
     # Brute-force our way through GLIBCXX_* symbols to discover which version we're linked against
-    hdl = Libdl.dlopen(first(libstdcxx_paths))
+    hdl = Libdl.dlopen(first(libstdcxx_paths))::Ptr{Cvoid}
```

Comment:
Seems like this should be inside `dlopen`?

Reply:
dlopen optionally can take throw_error=false as a kwarg and return `nothing`